src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -54,7 +55,7 @@ public static IHostBuilder UseInvocationLifetime(this IHostBuilder host, Action<
         public static CliCommand UseCommandHandler<THandler>(this CliCommand command)
             where THandler : CliAction
         {
-            command.Action = CommandHandler.Create(typeof(THandler).GetMethod(nameof(CliAction.InvokeAsync)));
+            command.Action = CommandHandler.Create(typeof(THandler).GetMethod(nameof(AsynchronousCliCommandAction.InvokeAsync)));
```

Comment:
This code is related to the `Hosting` package's dependency on `NamingConventionBinder`. Since neither of these packages is planned for GA at this time and the use of reflection will be completely replaced, it's unlikely we'll fix these.